Sources/TuistDependencies/Carthage/Models/CarthageVersionFile.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import TuistGraph
+
+/// A model that represents the `Carhtage` version file.
+/// Reference: https://github.com/Carthage/Carthage/blob/master/Documentation/VersionFile.md#version-files
+struct CarthageVersionFile: Decodable, Equatable {
+    enum CodingKeys: String, CodingKey {
+        case commitish
+        case iOS
+        case macOS = "Mac"
+        case watchOS
+        case tvOS
+    }
+
+    let commitish: String
+    let iOS: [Product]
+    let macOS: [Product]
+    let watchOS: [Product]
+    let tvOS: [Product]
+
+    /// Returns all products.
+    var allProducts: [Product] {
+        iOS + macOS + watchOS + tvOS
+    }
+}
+
+// MARK: - Models
+
+extension CarthageVersionFile {
+    struct Product: Decodable, Equatable {
+        let swiftToolchainVersion: String
+        let hash: String
+        let name: String
+        let container: String
+        let identifier: String
+
+        /// Returns architectures the product is built for.
+        var architectures: [BinaryArchitecture] {
+            // example identifier: `ios-arm64_i386_x86_64-simulator`
+            identifier
+                .components(separatedBy: "-")[1]
+                .replacingOccurrences(of: "x86_64", with: "x8664")
+                .replacingOccurrences(of: "arm64_32", with: "arm6432")
+                .components(separatedBy: ["_"])
+                .map { $0 == "x8664" ? "x86_64" : $0 }
+                .map { $0 == "arm6432" ? "arm64_32" : $0 }
```

Comment:
this seems redundant with the `replacingOccurrences` functions above?

Reply:
I have to do that to escape `arm64_32` and `x86_64` to split the string by `_` 

once string is split I have to restore original format to meet [TuistGraph.BinaryArchitecture](https://github.com/tuist/tuist/blob/ba28383dc89aab1d411002c0303fe794e24db767/Sources/TuistGraph/Models/BinaryArchitecture.swift#L3)'s raw values.

Reply:
Right, the separator of `_` used by carthage is not the best. Let's keep it as-is